webapp/src/components/surveyDateSelector/surveyDateSelector.tsx

```diff
@@ -1,36 +1,41 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import React, {useCallback} from 'react';
-import {format} from 'date-fns';
+import React, {useEffect, useState} from 'react';
 
 import DatePicker from 'components/common/datePicker';
 
 import './style.scss';
 
+const dateFormattingOptions: Intl.DateTimeFormatOptions = {
+    year: 'numeric',
+    month: '2-digit',
+    day: 'numeric',
+};
+
 export type Props = {
-    value: Date
-    onChange?: (value: Date) => void
+    value: Date;
+    onChange: (value: Date) => void;
 };
 
 const SurveyDateSelector = ({value, onChange}: Props) => {
-    const onSelectHandler = useCallback((newValue: Date) => {
-        if (onChange) {
-            onChange(newValue);
-        }
-    }, [onChange]);
+    const [formatedDate, setFormattedDate] = useState<string>('');
```

Comment:
Nit: Don't think you need the generic here if you're specifying a default value.

Reply:
Yeah but I just specify it anyways everywhere out of habit :P